File: CameraController/Core/CameraControl/Camera.cs
using System.Threading;
using System.Threading.Tasks;

namespace CameraControl
{
    public abstract class Camera
    {
        public abstract string Model { get; }
        
        public abstract string Port { get; }

        // TODO: Where cancellation token?
        public abstract Task<string> CaptureImage(string path, string filename);

        public abstract Task<byte[]> CapturePreview(CancellationToken ct);
    }
}

File: CameraController/Presentation/gui/src/components/app/Camera.ts
export class Camera {
  private model: string;
  private port: string;

  constructor(model: string, port: string) {
    this.model = model;
    this.port = port;
  }

  get Model(): string {
    return this.model;
  }

  get Port(): string {
    return this.port;
  }
}
